From b5fbd992cec3768460f2c345727741b160f562b1 Mon Sep 17 00:00:00 2001
From: Ning Sun <ning.sun@intel.com>
Date: Sun, 18 Feb 2018 08:08:30 -0800
Subject: [PATCH 07/22] Fix a null pointer dereference bug when Intel TXT is
 disabled. The issue can be found with Intel TXT being disabled and using
 tboot to boot into Linux Kernel without launching TXT environment, tboot goes
 into an endless loop.

Signed-off-by: Bo Adler <thumper@fb.com>
Reviewed-by: Ning Sun <ning.sun@intel.com>
(cherry picked from commit 1a5bdc2b07990c3792da8e1cf359a410414ec8ce)
---
 tboot/common/tb_error.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tboot/common/tb_error.c b/tboot/common/tb_error.c
index e9e8244..e02e9f1 100644
--- a/tboot/common/tb_error.c
+++ b/tboot/common/tb_error.c
@@ -167,8 +167,8 @@ bool write_tb_error_code(tb_error_t error)
 {
     struct tpm_if *tpm = get_tpm();
     const struct tpm_if_fp *tpm_fp = get_tpm_fp();
-
-    if ( !tpm || no_err_idx )
+    
+    if ( !tpm || !tpm_fp || no_err_idx )
         return false;
 
     if ( !tpm_fp->nv_write(tpm, tpm->cur_loc, tpm->tb_err_index, 0,
-- 
2.16.1

